### PR TITLE
Loadable extensions

### PR DESCRIPTION
**Pull request by mlyle**
Ready for preliminary review

### CODE REVIEW

**Comment by mlyle** [comment]
@ernieift  --- this might interest you-- it supersedes PicoC in dRonin.

File: flight/module/Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ARM_SDK_PREFIX = $(ARM_SDK_DIR)/bin/arm-none-eabi-
+TCHAIN_PREFIX  = $(ARM_SDK_PREFIX)
+CC             = $(TCHAIN_PREFIX)gcc
+OBJCOPY        = $(TCHAIN_PREFIX)objcopy
+
+SRC=modtest.c
+
+MCU = cortex-m4
+
+FLIGHTPATH = /home/mlyle/dRonin/flight
```
Comment (peabody124) [inline flight/module/Makefile]: shouldn't be hardcoded

File: flight/module/Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
+TOOLS_DIR      = /home/mlyle/dRonin/tools
```
Comment (peabody124) [inline flight/module/Makefile]: shouldn't be hardcoded

Comment (mlyle) [reply flight/module/Makefile]: Thx for review  :)

Yah-- The build system is completely wrong (in description as WIP and the individual commit comments as "DONTCOMMIT")  There's a lot of little things to figure out about how to find includes.

The modules stuff has to reside out of tree, and ... shouldn't depend on so many include dirs, etc.  It should also be considerably less complicated than our overall build system.  So there's work to be done on how the build system works for them.

**Comment by mlyle** [comment]
Overall I think the plan here is to let this stall awhile, and get a few commits like #1927 through that clean up the build system so that the module builds can ultimately be cleaner / nicer.

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
Rebased.  Need to work on build system, and export a few more calls so it's minimally useful.

**Comment by mlyle** [comment]
@tracernz -- can you please review -- this is an acceptable / useful state.

**Review by tracernz — APPROVED** [review]
I have read the early commits several times, and the most recent (fairly trivial) commits once. Looks fine to me.

**Pull request merged**